### PR TITLE
Fix some warnings found by valgrind

### PR DESCRIPTION
**Pull request by DolceTriade**

### CODE REVIEW

**Comment by bmorel** [comment]
No lie: I would have appreciated having valgrind's output. I should probably look at the code a bit more before being ok with this (but this clearly looks like it should work)

**Review by necessarily-equal — COMMENTED** [review]
untested, looks ok to me

File: src/shared/parse.cpp [hunk 1]
```diff
@@ -3991,6 +3991,24 @@ int Parse_AddGlobalDefine( const char *string )
...
 
+/*
+===============
+Parse_FreeGlobalDefines
+
+Frees all global defines.
+===============
+*/
+void Parse_FreeGlobalDefines()
+{
```
[the remark this answers is not in the document]
Comment (DolceTriade) [reply src/shared/parse.cpp:4002]: Done

**Review by necessarily-equal — APPROVED** [review]
didn't read it all in depth again, still LGTM witout a test

**Pull request merged**